### ISSUE TITLE
Set default retention policy in InfluxDB

### ISSUE THREAD

**Issue by vishh** [kubernetes/heapster#566]
We should set default retention policy in InfluxDB to be 5 days. Any data that needs to persist longer should be moved to a different database using continuous queries and separate retention policies.

Any thoughts @mwielgus @piosz ?

**Comment by vishh** [comment]
There are quite a few issues around disk usage of InfluxDB. I'm hoping 5 days will work for most deployments.

**Comment by piosz** [comment]
SGTM

**Comment by Pendoragon** [comment]
Any updates on this?

**Comment by moserke** [comment]
Seems like the retention policy name, like db name, should be configurable through the sink.  At sink startup should be able to pass in a retention policy.  Then users can create retention policies in influx themselves for their own cases.

Alternatively could expose the duration value and let the sink build the policy on the fly configured to that duration.  This to me seems more flexible.  The default value of the duration can be 5 days, but users can set the duration to a different number if they so deem.

Thoughts?

**Comment by vishh** [comment]
A configurable value SGTM.

**Comment by dcolens** [comment]
- by retention policy does it mean keeping the data x days after the container is deleted or x-days after the data is created ?
- in the case of short-lived container, it would be nice to have an aggressive retention policy. What would be a way to specify different retention policies based on the container/label ?

**Comment by vishh** [comment]
> by retention policy does it mean keeping the data x days after the container is deleted or x-days after the data is created ? From the influxdb doc it sounds like it is the later.

It is the latter. 

> in the case of short-lived container, it would be nice to have an aggressive retention policy. What would be a way to specify different retention policies based on the container/label ?

We should let users configure the retention policies because one solution might not work for everyone.

**Comment by Malmee** [comment]
Is this retention policy implemented? I am using heapster to find the idle containers in the Kubernetes cluster. I want to examine the resource details of the containers in Kubernetes cluster and detect the idle containers. But I can access only data up to 15 minutes. Does the data get deleted or is there anyway to get data upto 5 days?

**Comment by piosz** [comment]
Heapster by default provides 15 minutes of data in its API. In theory this duration could be extended, but it would cause Depending on the configuration Heapster can export data to InfluxDB (or many others metric backends) and there should remain for a longer period of time. I suggest using the second option.

**Comment by Crapworks** [comment]
Any news here? That would be pretty valuable to us, since we are spinning k8s clusters in openstack und start an influxdb as a heapster backend in a pod. Right now I just fire up influxdb with a custom config. the database that heapster is creating has a retention policy for 168h (7 days). I would like to keep the data for 30days. Is there still work on this?

**Comment by piosz** [comment]
We are happy to accept a contribution, though we have higher priority task to do.

**Comment by mpvaniersel** [comment]
It looks like I started working on this in parallel with @bavarianbidi. I can rework this in whichever way necessary to get this upstreamed.

**Comment by mpvaniersel** [comment]
I'm comparing the two pull requests to see what I can learn here - The one of @bavarianbidi makes it possible to override the retentionpolicy selected when sending a batch of points (i.e. makes it possible to put measurements under a non-default policy), but still requires you to somehow create and set that non-default retention policy elsewhere. On the other hand my pull request is designed to override the duration on the default retention policy, covering all measurements. Disclaimer: I'm pretty new to go, influxdb and heapster so I might be misunderstanding things.
